clang/lib/Sema/AnalysisBasedWarnings.cpp

```diff
@@ -27,6 +27,8 @@
 #include "clang/AST/StmtObjC.h"
 #include "clang/AST/StmtVisitor.h"
 #include "clang/AST/Type.h"
+#include "clang/ASTMatchers/ASTMatchFinder.h"
```

Comment:
FYI, these two headers require a cmake change:

```
diff --git a/clang/lib/Sema/CMakeLists.txt b/clang/lib/Sema/CMakeLists.txt
index a96439df6642..278c7c7394ab 100644
--- a/clang/lib/Sema/CMakeLists.txt
+++ b/clang/lib/Sema/CMakeLists.txt
@@ -80,6 +80,7 @@ add_clang_library(clangSema
   LINK_LIBS
   clangAPINotes
   clangAST
+  clangASTMatchers
   clangAnalysis
   clangBasic
   clangEdit
```

With this small change I was able to build the branch of your work merged with ours, no other adaptations necessary :)

Reply:
Er, Sema depending on ASTMatchers is *probably* fine, but I’d like someone more familiar w/ our library layering to confirm whether this is ok.

Reply:
To be less prescriptive with the solution:

I was getting linker errors saying that the symbols found in the ASTMatchers headers (used later in this file) were missing. I fixed it with this CMake change. Let me know if anyone needs more info and I'll re-generate the error.

Reply:
No yeah, that part makes sense; I’d assume that you need to link to clangASTMatchers to be able to use AST matchers. I’m just saying I’m not sure whether we want to use AST matchers in Sema as there doesn’t seem to be a precedent for it—though I’m not familiar enough w/ the analysis part of Sema to know whether we’ve just never needed them or whether there’s another reason why you might not want to use them (afaik AST matchers are one of those components like libclang that are maintained on a best-effort basis, but I could be wrong).

Reply:
The ASTMatcher is only used in some debug-only code. It uses an AST traversal to find all the Decls to verify and then compares it against Sema's `DeclsWithEffectsToVerify`. I'm convinced now that Sema isn't missing anything so I'll remove the dependency.